test/metabase/lib/test_util.cljc

```diff
@@ -264,6 +280,12 @@
                                        (mapv #(dissoc % :id :table-id)))}))])))
         (meta/tables)))
 
+(defn metadata-provider-with-mock-card [card]
+  (lib/composed-metadata-provider
+    meta/metadata-provider
+    (mock-metadata-provider
+      {:cards [card]})))
```

Comment:
I wrote almost this exact same utility in 

https://github.com/metabase/metabase/pull/33429/files#diff-1da1a58aabf037690d9d1a71c9254ccfdb75eecb7b39f1888f3644ed0d37441aR10-R26

Reply:
There's a lot of pieces of this branch that will need consolidating. Most of the `lib.js` functionality is partially duplicated from filters work, etc. I'll clean it up in a following PR.